tools/autograd/gen_python_functions.py

```diff
@@ -497,6 +508,30 @@ def emit_single_dispatch(declaration, is_python_method, output_gap=0):
     dispatch_callee = get_dispatch_callee(declaration)
     dispatch_args = get_op_args(declaration, {name: name for name, _ in argmap.items()})
 
+    # This is a hack.
+    # Please see [Add requires_grad to native_functions.yaml and potentially to TensorOptions object]
+    # In the tracking issue: https://github.com/pytorch/pytorch/issues/30405
+    #
+    # Adding requires_grad to formals and args.
+    if check_is_factory_or_like_or_new_function(declaration) and not is_tensor_method(declaration):
```

Comment:
Can we move this logic into [make_python_binding_args](https://github.com/pytorch/pytorch/blob/master/tools/autograd/gen_python_functions.py#L1302)? That's where similar transforms happen currently, so it'll keep things more centralized and discoverable (and leave this last-stop codegen stage cleaner and more robust). 

Ooo it'd be especially nice if you could port that code to use your new `check_is_factory_or_like_or_new`, which [this logic](https://github.com/pytorch/pytorch/pull/30983/files#diff-d5f0cc9f3f43b53571f5be87e91587f3R1422) still duplicates in the current PR.

ofc LMK if there's some reason the hack needs to get plugged in here, but I'm not seeing anything that would prevent it from being colocated with the other similar logic.